### ISSUE TITLE
Incorrect definition of `wcsstr` for older glibc depending on include order

### ISSUE THREAD

**Issue by rupprecht** [llvm/llvm-project#62638]
This snippet fails to compile, as expected:
```c++
#include <cwchar>
#include <iosfwd>
void func() {
  wchar_t* v1;
  const wchar_t* cv2 = L"/";
  v1 = wcsstr(cv2, L"/"); // expected-error {{assigning to 'wchar_t *' from 'const wchar_t *' discards qualifiers}}
}
```

D148542 adds another path to including the system `wchar.h` without defining `__CORRECT_ISO_CPP_WCHAR_H_PROTO` as we do elsewhere, and older versions of glibc did not declare `__CORRECT_ISO_CPP_WCHAR_H_PROTO` for itself correctly when clang is the host compiler. In that situation, changing the include order causes it to unexpectedly compile:

```c++
#include <iosfwd>  // This and the next line are the only differences
#include <cwchar>
void func() {
  wchar_t* v1;
  const wchar_t* cv2 = L"/";
  v1 = wcsstr(cv2, L"/"); // Unexpectedly no error here
}
```

D150015 fixes this.

**Comment by philnik777** [comment]
Fixed by 0ec57bf7b18e
